### PR TITLE
[19] FEATURE: 2FA with OTP for sign in

### PR DESCRIPTION
**Pull request by zrichard22**

### CODE REVIEW

**Review by HugoTourneur — DISMISSED** [review]

File: mobile-app/components/business/PrivacySettings.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+              <Switch.Thumb />
+            </Switch>
+          </SettingsListItem>
+          <SettingsItemPrivacySearchEmailPhone
+            isEnabled={settings?.allowSearchEmailPhone}
+          />
+          <SettingsItemPrivacy2FA isActivated={settings?.has2FA} />
+          <SettingsListItem label='Password'>
+            <Button>
+              Change
```
Comment (Lucas-Senoville) [inline mobile-app/components/business/PrivacySettings.tsx:56]: Use i18n for text

File: mobile-app/components/business/SettingsListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const SettingsListItem = ({ label, children }: SettingsListItemProps) => (
+  <View
+    flex={1}
+    flexDirection='row'
+    alignItems='center'
+    paddingHorizontal='$1'
+    height='$10'
+  >
+    <View flexGrow={1}>
+      <Text>{label}</Text>
```
Comment (Lucas-Senoville) [inline mobile-app/components/business/SettingsListItem.tsx:18]: Use i18n for text

File: mobile-app/components/sheet/ActiveOtp.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    setTimeout(() => {
+      setPressed(false)
+    }, 5000)
+  }
+
+  return (
+    <View padding='$4' flex={1} alignItems='center' gap='$8'>
+      <Text type='title'>2 Factor Authentication</Text>
+      <Text>
+        To use the 2FA, you need to past the code generated by the app in the
```
Comment (Lucas-Senoville) [inline mobile-app/components/sheet/ActiveOtp.tsx:25]: Use i18n for text

File: mobile-app/components/sheet/ActiveOtp.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      <View onPress={handlePastClipboard}>
+        <Input
+          placeholder='Code'
+          readOnly
+          value={code}
+          borderRadius='$1'
+          width='50%'
+        />
+      </View>
+      {pressed && <Text>Code copied to clipboard</Text>}
```
Comment (Lucas-Senoville) [inline mobile-app/components/sheet/ActiveOtp.tsx:37]: Use i18n for text

File: mobile-app/screens/otp.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        setLogin(true)
+        router.replace('/')
+      }
+    } catch (err) {
+      if (axios.isAxiosError(err)) {
+        if (err.response) {
+          setError(err.response.data)
+        }
+      } else {
+        setError('An error occurred')
```
Comment (Lucas-Senoville) [inline mobile-app/screens/otp.tsx:49]: Use i18n for text

File: mobile-app/screens/otp.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+  const handleChangeCode = (text: string) => {
+    setCode(text)
+  }
+
+  return (
+    <CustomSafeArea>
+      <View alignItems='center' justifyContent='center' height='100%' gap='$8'>
+        <Text type='title'>OTP Check</Text>
+        <Text>Enter the OTP code from your OTP App</Text>
```
Comment (Lucas-Senoville) [inline mobile-app/screens/otp.tsx:62]: Use i18n for text

File: mobile-app/screens/otp.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        <Input
+          placeholder='OTP Code'
```
Comment (Lucas-Senoville) [inline mobile-app/screens/otp.tsx:64]: Use i18n for text

File: mobile-app/screens/otp.tsx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+          borderRadius='$2'
+          width='70%'
+          onChangeText={handleChangeCode}
+        />
+        <View width='70%'>
+          <CustomButton
+            color='primary'
+            onPress={handleSubmitOTP}
+            text='Check'
```
Comment (Lucas-Senoville) [inline mobile-app/screens/otp.tsx:73]: Use i18n for text

Comment (Lucas-Senoville) [inline mobile-app/screens/otp.tsx:73]: Use i18n for text

**Review by Lucas-Senoville — CHANGES_REQUESTED** [review]

File: mobile-app/components/business/SettingsListItem.tsx (hunk 2 above)
Comment (zrichard22) [reply mobile-app/components/business/SettingsListItem.tsx:18]: This is a variable and no a fixed text

**Review by HugoTourneur — APPROVED** [review]

**Review by Lucas-Senoville — APPROVED** [review]

**Pull request merged**